core/controllers/editor.py

```diff
@@ -266,7 +267,10 @@ def put(self, exploration_id: str) -> None:
                 exp_services.update_exploration(
                     self.user_id, exploration_id, change_list, commit_message
                 )
-            elif can_voiceover and changes_are_mergeable:
+            else:
+                # Unauthorized users are blocked by decorator.
+                # Assertion serves to make coverage and intent explicit.
+                assert can_voiceover and changes_are_mergeable
```

Comment:
Are we sure that in the else condition we will always have `can_voiceover` and `changes_are_mergeable`?